src/main/java/teammates/storage/search/SearchManagerFactory.java

```diff
@@ -0,0 +1,28 @@
+package teammates.storage.search;
+
+/**
+ * Factory that returns search manager implementation.
+ */
+public final class SearchManagerFactory {
+
+    private static SearchManager instance;
+
+    private SearchManagerFactory() {
+        // prevents initialization
+    }
+
+    public static SearchManager getSearchManager() {
+        return instance;
+    }
+
+    /**
+     * Registers the search service into the factory.
+     */
+    @SuppressWarnings("PMD.NonThreadSafeSingleton") // ok to ignore as method is only invoked at application startup
+    public static void registerSearchManager(SearchManager searchManager) {
+        if (instance == null) {
+            instance = searchManager;
```

Comment:
Any reason for `null` check? Intuitively the registration should overwrite whatever the old instance.

Reply:
The instance is final and should not be changed once set in the entire application lifecycle. Since using `final` obviously will not work here, this is the best alternative.